sql/hive/src/test/scala/org/apache/spark/sql/hive/CompressionCodecSuite.scala

```diff
@@ -0,0 +1,321 @@
+        |$partitionCreate
+        |STORED AS $format
+        |LOCATION '${rootDir.toURI.toString.stripSuffix("/")}/$tableName'
+        |$tblProperties
+      """.stripMargin)
+  }
+
+  private def writeDataToTable(
+      tableName: String,
+      partition: Option[String]): Unit = {
+    val partitionInsert = partition.map(p => s"partition (p='$p')").mkString
+    sql(
+      s"""
+        |INSERT INTO TABLE $tableName
+        |$partitionInsert
+        |SELECT * FROM table_source
+      """.stripMargin)
+  }
+
+  private def getTableSize(path: String): Long = {
+    val dir = new File(path)
+    val files = dir.listFiles().filter(_.getName.startsWith("part-"))
+    files.map(_.length()).sum
+  }
+
+  private def getTablePartitionPath(dir: File, tableName: String, partition: Option[String]) = {
+    val partitionPath = partition.map(p => s"p=$p").mkString
+    s"${dir.getPath.stripSuffix("/")}/$tableName/$partitionPath"
+  }
+
+  private def getUncompressedDataSizeByFormat(
+      format: String, isPartitioned: Boolean): Long = {
+    var totalSize = 0L
+    val tableName = s"tbl_$format"
+    val codecName = normalizeCodecName(format, "uncompressed")
+    withSQLConf(getSparkCompressionConfName(format) -> codecName) {
+      withTempDir { tmpDir =>
+        withTable(tableName) {
+          createTable(tmpDir, tableName, isPartitioned, format, Option(codecName))
+          val partition = if (isPartitioned) Some("test") else None
+          writeDataToTable(tableName, partition)
+          val path = getTablePartitionPath(tmpDir, tableName, partition)
+          totalSize = getTableSize(path)
+        }
+      }
+    }
+    assert(totalSize > 0L)
+    totalSize
+  }
+
+  private def checkCompressionCodecForTable(
+      format: String,
+      isPartitioned: Boolean,
+      compressionCodec: Option[String])
+      (assertion: (String, Long) => Unit): Unit = {
+    val tableName = s"tbl_$format$isPartitioned"
+    withTempDir { tmpDir =>
+      withTable(tableName) {
+        createTable(tmpDir, tableName, isPartitioned, format, compressionCodec)
+        val partition = if (isPartitioned) Some("test") else None
+        writeDataToTable(tableName, partition)
+        val path = getTablePartitionPath(tmpDir, tableName, partition)
+        val relCompressionCodecs = getTableCompressionCodec(path, format)
+        assert(relCompressionCodecs.length == 1)
+        val tableSize = getTableSize(path)
+        assertion(relCompressionCodecs.head, tableSize)
+      }
+    }
+  }
+
+  private def checkTableCompressionCodecForCodecs(
+      format: String,
+      isPartitioned: Boolean,
+      convertMetastore: Boolean,
+      compressionCodecs: List[String],
+      tableCompressionCodecs: List[String])
+      (assertionCompressionCodec: (Option[String], String, String, Long) => Unit): Unit = {
+    withSQLConf(getConvertMetastoreConfName(format) -> convertMetastore.toString) {
+      tableCompressionCodecs.foreach { tableCompression =>
```

Comment:
You are setting compressionCodecs and tableCompressionCodecs to the same one. I think we can set it to different ones to know which one is being used.

Reply:
Shouldn't we test the precedence of compressionCodecs in effect? Here I just want to verify the precedence.